ArduCopter/RC_Channel.cpp

```diff
@@ -151,12 +151,12 @@ void RC_Channel_Copter::do_aux_function(const aux_func_t ch_option, const aux_sw
 
         case AUX_FUNC::SIMPLE_MODE:
             // low = simple mode off, middle or high position turns simple mode on
-            copter.set_simple_mode(ch_flag == HIGH || ch_flag == MIDDLE);
+            copter.set_simple_mode((ch_flag == HIGH || ch_flag == MIDDLE)?Copter::SIMPLEMODE_ON:Copter::SIMPLEMODE_OFF);
             break;
 
         case AUX_FUNC::SUPERSIMPLE_MODE:
             // low = simple mode off, middle = simple mode, high = super simple mode
-            copter.set_simple_mode(ch_flag);
+            copter.set_simple_mode(static_cast<Copter::SimpleMode>(ch_flag));
```

Comment:
This doesn't look great; it's assuming there's a 1-1 correspondence between the `ch_flag` enumeration and the new enumeration.  I think a `switch` in here (even if it is nested :-( ) may be the best option.

Reply:
I will make a switch.